### ISSUE TITLE
🐛 Panic

### ISSUE THREAD

**Issue by jasha-hrp** [dandavison/delta#1857]
Delta crashed when I called `git diff`
Sorry I can't provide a screenshot / raw git output as I'm working on proprietary code.

- [ ] Please include the raw text output from git, so that we can reproduce the problem.
      (You can use `git --no-pager` to produce the raw text output.)
- [ ] A screenshot of Delta's output is often helpful also.

```
│    │                                                                …amp")  │    │                                                                …amp")
thread 'main' panicked at /Users/jasha/.cargo/registry/src/index.crates.io-6f17d22bba15001f/git-delta-0.18.1/src/wrapping.rs:469:9:
assertion `left == right` failed: syntax and diff wrapping differs (l) [*r*]
  left: (0, 1)
 right: (0, 2)
stack backtrace:
   0:        0x102d527d0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h41035ce174e31160
   1:        0x102d703f8 - core::fmt::write::h7e946826fce7616b
   2:        0x102d4f650 - std::io::Write::write_fmt::he3645adfefb23e4a
   3:        0x102d52628 - std::sys_common::backtrace::print::h2efe9ae66fda73dc
   4:        0x102d539a0 - std::panicking::default_hook::{{closure}}::hd27200b4fbd3bf40
   5:        0x102d5366c - std::panicking::default_hook::hb8656334461229c8
   6:        0x102d5424c - std::panicking::rust_panic_with_hook::h10171cf76e1aed15
   7:        0x102d53c88 - std::panicking::begin_panic_handler::{{closure}}::h9344de43a47cae21
   8:        0x102d52c54 - std::sys_common::backtrace::__rust_end_short_backtrace::h55013ada3ab9c4e8
   9:        0x102d539f8 - _rust_begin_unwind
  10:        0x102d89e64 - core::panicking::panic_fmt::h0b16bb09366e1f01
  11:        0x102d8a164 - core::panicking::assert_failed_inner::h9886fa19a1f8658e
  12:        0x102d7e9fc - core::panicking::assert_failed::h62bbe4faec7d578d
  13:        0x102aec764 - delta::wrapping::wrap_minusplus_block::wrap_syntax_and_diff::h47f27fe9d031e69c
  14:        0x102aea5ec - delta::wrapping::wrap_minusplus_block::h9b25f4e82e855b71
  15:        0x102adc544 - delta::paint::paint_minus_and_plus_lines::hfc3334bdd5401081
  16:        0x102ad9934 - delta::paint::Painter::paint_buffered_minus_and_plus_lines::hb0078759ad48923b
  17:        0x102b2f1b0 - delta::handlers::hunk::<impl delta::delta::StateMachine>::handle_hunk_line::h9b14d30dd1f18471
  18:        0x102b294c4 - delta::delta::delta::hd04f0016a944a858
  19:        0x102ae01a8 - delta::run_app::h674815f1d32de915
  20:        0x102adea44 - delta::main::h0d3cd62062917ade
  21:        0x102b33188 - std::sys_common::backtrace::__rust_begin_short_backtrace::hc8b80ffaa75ddde1
  22:        0x102b0a578 - std::rt::lang_start::{{closure}}::h1300defa2b83e158
  23:        0x102d485e8 - std::rt::lang_start_internal::h27a134f18d582a1e
  24:        0x102ae1e94 - _main
11:48:09 .venv jasha-hrp/EP-10793_r28_docs [$!?] on ☁️   quantdev (us-east-1) ~/digital-pb/backend/code $ delta --version
delta 0.18.1
```

**Comment by jasha-hrp** [comment]
Resizing the terminal (making it wider) alleviated the issue.

**Comment by th1000s** [comment]
If you obtained the binary via `cargo install` then please try `cargo install --locked git-delta` [via shaicoleman in #1726](https://github.com/dandavison/delta/issues/1726#issuecomment-2339307377). It seems a dependency only works as expected when using the _exact_ version that `delta` specifies. A normal `cargo install` does not do that.

**Comment by th1000s** [comment]
Should be fixed in 0.18.2 and work without `--locked` now.
